app/controllers/categories_controller.rb

```diff
@@ -2,9 +2,11 @@
 
 class CategoriesController < ApplicationController
   decorates_assigned :category, :books
+  before_action :show_filter, only: %i[index show]
 
   def index
     @pagy, @books = pagy(BookSorter.call(sort_params))
+    @filter = params[:status] ? params[:status].to_s : BookSorter::DEFAULT_FILTER
```

Comment:
show_filter method works here, so, why this line appears again?

Reply:
removed this line as we have it defined as a before_action method